### PR TITLE
Support Rust SDK by gRPC-rs

### PR DESCRIPTION
**Pull request by thara**
( This PR is restarted from https://github.com/GoogleCloudPlatform/agones/pull/167 to keep things clean. )

[Rust](https://www.rust-lang.org/en-US/) is a system programming language that runs fast, has memory safety.
It is good for game server implementation too.

Core implementations of the SDK functions are generated by [gRPC-rs](https://crates.io/crates/grpcio) and located to `sdks/rust/src/grpc` directory.

### CODE REVIEW

**Comment by markmandel** [comment]
At first pass, this looks pretty awesome! I'll take a deeper look as soon as I can, but a few things to get this to completion:

I noticed that there is no Dockerfile, or gameserver.yaml for the example? We'll need those so people can test this out. Maybe we can do a [multistage-build](https://docs.docker.com/develop/develop-images/multistage-build/) so people can test it without having to install anything? (We could actually do this for lots of our examples).

Also, can we have a README for the example, explaining what is does, how to build it, etc?

We will want to edit the [sdk docs](https://github.com/GoogleCloudPlatform/agones/tree/master/sdks) to include a reference to the Rust SDK. Which may mean we need some kind of Rust docs. Not sure if there is an auto-generated version like godoc we can use, or whether this should be just some hand written markdown?

Looking forward to taking this for a spin!

**Comment by thara** [comment]
@markmandel  I added gameserver.yml for running minikude. 
But I found that the example make the pod to make CrashLoopBackOff status.
I'm going to fix it.

**Comment by thara** [comment]
@Kuqd I added Rust SDK generation to gen-gameservers-sdk-grpc task.
(I missed versions of SDK generation tools and bump up versions of them)

I'm still working to fix the [CrashLoopBackOff bug](https://github.com/GoogleCloudPlatform/agones/pull/230#issuecomment-394064167) in rust-sample.
Maybe I made a mistake on using health API.

**Comment by Kuqd** [comment]
Don't hesitate to ask question on slack so we can help you troubleshoot, is the simple-udp is working in your cluster ? By this I mean is only your example is in crashloop, did you try other example ? It could be a wrong installation...

**Comment by markmandel** [comment]
This is looking awesome! Love the new docs, and the integration into the build system!

File: examples/rust-simple/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+RUN mkdir -p /usr/src/cmake \
+  && curl -fSLO https://cmake.org/files/${CMAKE_MINOR_VERSION}/cmake-${CMAKE_FULL_VERSION}.tar.gz \
+  && curl -fSLO https://cmake.org/files/${CMAKE_MINOR_VERSION}/cmake-${CMAKE_FULL_VERSION}-SHA-256.txt \
+  && sha256sum -c cmake-${CMAKE_FULL_VERSION}-SHA-256.txt 2>&1 | grep OK \
+  && tar xf cmake-${CMAKE_FULL_VERSION}.tar.gz -C /usr/src/cmake --strip-components=1 \
+  && rm -f cmake-${CMAKE_FULL_VERSION}.* \
+  && cd /usr/src/cmake \
+  && ./configure && make -j$(nproc) && make install
+
+## Go
```
Comment (markmandel) [inline examples/rust-simple/Dockerfile]: Love the multi-stage builder image - we should do that more.

Out of curiosity, why do we need Go here?

File: examples/rust-simple/gameserver.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  # generate a unique name
+  # will need to be created with `kubectl create`
+  generateName: rust-simple-
+spec:
+  containerPort: 7654
+  template:
+    spec:
+      containers:
+      - name: rust-simple
+        image: rust-simple-server:0.1
```
Comment (markmandel) [inline examples/rust-simple/gameserver.yaml]: This should be:
`gcr.io/agones-images/rust-simple-server:0.1`

As once this gets merged, I'll build this, and push the image up to the public registry :+1:

File: examples/rust-simple/Dockerfile (hunk 1 above)
Comment (thara) [reply examples/rust-simple/Dockerfile]: > Out of curiosity, why do we need Go here?

This Rust SDK depends [gRPC-RS](https://github.com/pingcap/grpc-rs).
It needs Go for TLS encryption and some authentication mechanism. 
See also https://github.com/pingcap/grpc-rs#prerequisites.

**Comment by thara** [comment]
@Kuqd Thanks you for your concern. 
The simple-udp is working correctly in my minikube cluster on my machine :)

**Comment by markmandel** [comment]
I just looked - can we rebase this down to a single commit? Once that's done, we should merge this in!

**Comment by thara** [comment]
I fixed the [CrashLoopBackOff bug](https://github.com/GoogleCloudPlatform/agones/pull/230#issuecomment-394064167) in rust-sample.

**Comment by thara** [comment]
@markmandel I rebased to a single commit.
(I left the old commit logs in [thara/support-rust-sdk-by-grpcrs-backup](https://github.com/thara/agones/tree/support-rust-sdk-by-grpcrs-backup) in just for the record)

**Comment by Kuqd** [comment]
sorry I merge a PR in between, so I rebase your branch, will merge asap.

**Pull request merged**